### PR TITLE
update builds api: allow status filter

### PR DESCRIPTION
**Pull request by sengineer0**
Ref: https://github.com/biothings/biothings_studio/issues/126

### CODE REVIEW

File: biothings/hub/databuild/builder.py [hunk 1]
```diff
@@ -1271,10 +1271,17 @@ def archive_merge(self, merge_name):
...
+    def get_query_for_list_merge(self, only_archived, status=None):
         q = {"archived": {"$exists": 0}}
         if only_archived:
             q = {"archived": {"$exists": 1}}
+        if status:
+            if status == "success":
+                q["jobs.status"] = {
+                    "$not": {"$in": ["failed", "canceled"]}
+                }
+            if status == "failed":
```
Comment (newgene) [inline biothings/hub/databuild/builder.py]: @sengineer0 if we have a status value of "canceled", we should support filtering on it too, right?

Comment (sengineer0) [reply biothings/hub/databuild/builder.py]: @newgene  right. I will add "canceled" status soon

Comment (sengineer0) [reply biothings/hub/databuild/builder.py]: @newgene it's updated

Comment (newgene) [reply biothings/hub/databuild/builder.py]: @sengineer0 that works, but I kind of don't like the parallel `if` blocks here. using `elif` and `else` instead here? and you can probably merge both "failed" and "canceled" cases into just one code block I think? Anyway, just need a bit of optimization here.

Comment (sengineer0) [reply biothings/hub/databuild/builder.py]: @newgene  let's me see

Comment (sengineer0) [reply biothings/hub/databuild/builder.py]: @newgene  it's updated

**Pull request merged**